code/datums/traits/good.dm

```diff
@@ -37,6 +37,20 @@
 	lose_text = "<span class='danger'>You no longer feel like drinking would ease your pain.</span>"
 	medical_record_text = "Patient has unusually efficient liver metabolism and can slowly regenerate wounds by drinking alcoholic beverages."
 
+/datum/quirk/drunkhealing/on_process()
+	var/mob/living/carbon/C = quirk_holder
+	if(C.drunkenness >= 6)
```

Comment:
this isn't really needed considering you haven't given anything in your switch statement to function under 6.